logbook-core/src/main/java/org/zalando/logbook/core/attributes/CompositeAttributeExtractor.java

```diff
@@ -0,0 +1,78 @@
+package org.zalando.logbook.core.attributes;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+import org.zalando.logbook.HttpRequest;
+import org.zalando.logbook.HttpResponse;
+import org.zalando.logbook.attributes.AttributeExtractor;
+import org.zalando.logbook.attributes.HttpAttributes;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+@API(status = EXPERIMENTAL)
+@AllArgsConstructor
+@EqualsAndHashCode
+@Slf4j
+public final class CompositeAttributeExtractor implements AttributeExtractor {
+
+    private final List<AttributeExtractor> attributeExtractors;
+
+    @Nonnull
+    @Override
+    public HttpAttributes extract(final HttpRequest request) {
+        final Map<String, Object> map = new HashMap<>();
+        for (final AttributeExtractor attributeExtractor : attributeExtractors) {
+            map.putAll(safeRequestExtractor(attributeExtractor, request));
+        }
+        return new HttpAttributes(map);
+    }
+
+    @Nonnull
+    @Override
+    public HttpAttributes extract(final HttpRequest request, HttpResponse response) {
+        final Map<String, Object> map = new HashMap<>();
+        for (final AttributeExtractor attributeExtractor : attributeExtractors) {
+            map.putAll(safeRequestExtractor(attributeExtractor, request, response));
+        }
+        return new HttpAttributes(map);
+    }
+
+    @Nonnull
+    private HttpAttributes safeRequestExtractor(final AttributeExtractor attributeExtractor,
+                                                final HttpRequest request) {
+        try {
+            return attributeExtractor.extract(request);
+        } catch (Exception e) {
+            log.trace(
```

Comment:
Maybe instead of catching, logging and falling back to a default value here, what about collecting all exceptions in a single exception (maybe using suppressed expections?) and throw that one instead?

Reply:
Do we actually want to continue in case an extractor fails?

Reply:
Are we doing the same with a sink, formatter or writer right now?

Reply:
My line of thinking about how to handle extractor failure was as follows: Assume we have multiple extractors, one of them being a JWT subject extractor, and an HTTP request has a malformed bearer token. While the extractor cannot extract the JWT subject claim (and thus throws an exception), we may be interested in extracting other attributes via the other extractors.

I'll look at `CompositeSink` to check whether it follows a similar logic.

Reply:
> Do we actually want to continue in case an extractor fails?

Yes, maybe some extractors succeed. I'd like to make it as robust as possible, and allow Logbook to extract as much information as it can. This way, extractors may fail independently of each other, allowing other successful ones to continue.

> Maybe instead of catching, logging and falling back to a default value here, what about collecting all exceptions in a single exception (maybe using suppressed expections?) and throw that one instead?

Fixed in 6fbff7a, with a little modification to the proposal: Instead of catching and throwing, we catch them and log all in one statement. This allows the non-failing ones to extract.

> Are we doing the same with a sink, formatter or writer right now?

`CompositeSink`, for example, uses fauxpas's `ThrowingConsumer`:

https://github.com/zalando/logbook/blob/8ad19a3e067b1d265cbf219714e21850f415f697/logbook-core/src/main/java/org/zalando/logbook/core/CompositeSink.java#L39-L41